framework/sandstone.cpp

```diff
@@ -2685,6 +2706,121 @@ static void warn_deprecated_opt(const char *opt)
+    if ((sApp->background_scan.init_timestamp == true) && (sApp->service_background_scan == true)) {
+        for (auto i = 0; i < sApp->background_scan.number_of_timestamps; i++) {
+            MonotonicTimePoint timestamp_init_val = 
+                    (now - sApp->background_scan.day_delay_between_tests);
+            sApp->background_scan.timestamp[i] = timestamp_init_val;
+        }
+    }
+}
+
+static void background_scan_update_load_threshold(void)
+{
+    MonotonicTimePoint now = MonotonicTimePoint::clock::now();
+
+    int64_t time_from_last_test = 
+        ((now - sApp->background_scan.timestamp[sApp->background_scan.timestamp_newest]) /
+        sApp->background_scan.number_seconds_in_hour).count();
+    // scale our idle threshold value from 0.2 base, to 0.8 after 12h
+    // every hour adds 0.05 to the threshold value
+    sApp->background_scan.load_idle_threshold = 
+        sApp->background_scan.load_idle_threshold_init +
+        (time_from_last_test * sApp->background_scan.load_idle_threshold_inc_val);
+
+    // prevent the idle threshold form rising above 0.8
+    if(sApp->background_scan.load_idle_threshold > sApp->background_scan.load_idle_threshold_max)
+        sApp->background_scan.load_idle_threshold = sApp->background_scan.load_idle_threshold_max;
+}
+
+static void background_scan_wait(void) 
+{
+    MonotonicTimePoint now = MonotonicTimePoint::clock::now();
+    // Don't run tests unless load is low or it's time to run a test anyway
+    while(1) {
+        // wait ~5 mins no matter what
+        sApp->delay_between_tests = sApp->background_scan.minimum_delay_between_tests;
+        sApp->background_scan.wait_deviation_percent = 10.0;
+        wait_delay_between_tests_with_deviation(
+                sApp->background_scan.wait_deviation_percent,
+                false);
+
+        now = MonotonicTimePoint::clock::now();
+
+        // If all the last 24 tests ran in less than the last 24h, don't run
+        // any tests at all
+
+        if (now < sApp->background_scan.timestamp[sApp->background_scan.timestamp_newest] + sApp->background_scan.day_delay_between_tests) {
+            sApp->delay_between_tests = sApp->background_scan.day_delay_between_tests;
+            sApp->background_scan.wait_deviation_percent = 0.1;
+                wait_delay_between_tests_with_deviation(
+                sApp->background_scan.wait_deviation_percent,
+                false);
+        }
+
+        background_scan_update_load_threshold();
+
+        // if the system is idle, run a test
+        if (system_is_idle(sApp->background_scan.load_idle_threshold))
+            break;
+
+        // if we haven't run *any* tests in the last x hours, run a test
+        // because of day/night cycles, 12 hours should help typical data center
+        // duty cycles.
+        if (now > (sApp->background_scan.timestamp[sApp->background_scan.timestamp_newest] + sApp->background_scan.half_day_delay_between_tests))
```

Comment:
Suggestion: rename the "day" variables to indicate their purpose, not their values. Name them something like "minimum time to force test running" or "max time for 24 tests".

Naming the variables after the values when you need to type more to get to the variable than the value they're holding is not ideal. You can just write `24h` and `12h` (literally) in this file.

Reply:
Good idea. We don't need to refactor them in the future if we decide to change requirements and for example force to run more tests earlier or later. We'll just need to modify values then.